py/picca/delta_extraction/utils.py

```diff
@@ -158,26 +158,26 @@ def find_bins(original_array, grid_array, wave_solution):
     grid_array.size with the bins correspondance
     """
     if wave_solution == "log":
-        pass
+        step = grid_array[1] - grid_array[0]
+        found_bin = np.floor((original_array - grid_array[0]) / step + 0.5).astype(np.int64)
     elif wave_solution == "lin":
-        original_array = 10**original_array
-        grid_array = 10**grid_array
+        aux = 10**grid_array[0]
+        step = 10**grid_array[1] - aux
+        found_bin = np.floor((10**original_array - aux) / step + 0.5).astype(np.int64)
```

Comment:
You don't need to floor.

Reply:
true, the `astype` is doing the floor on its own. I added a commit to remove them